sound/soc/sof/intel/hda-dai.c

```diff
@@ -151,14 +151,15 @@ static int hda_link_dma_params(struct hdac_ext_stream *stream,
 	return 0;
 }
 
-/* Send DAI_CONFIG IPC to the DAI that matches the dai_name and direction */
```

Comment:
commit message:
- (also in the code) `.ref_count` has a rather specific meaning in the kernel, I think, maybe just a `.use_count`?
- "we *no* longer need?"
- maybe it's just me, but what's the advantage of sending the `DAI_CONFIG` during FE DAI `.hw_params()` over that of the BE DAI? `DAI_CONFIG` is the BE DAI configuration, right? So, coupling the two seems logical to me...

Reply:
@lyakh its not an advantage to move DAI_CONFIG to FE DAI hw_params but rather a requirement. The order of hw_params is BE hw_params and then FE hw_params. And we set up the DAI widget only in FE hw_params. Therefore, we can no longer do DAI_CONFIG in BE hw_params because the DAI itself doesnt exist just yet